### ISSUE TITLE
Add Xcode generator

### ISSUE THREAD

**Issue by timursavchenko** [15knots/cmake4eclipse#156]
It's not possible using cmake4eclipse for generating build instructions for iOS, because Unix Makefiles and Ninja do not
work correctly with iOS targets.

Could you add, please, "Xcode" switch to supported generators in "Buildscript generator (-G)" dropdown list on "Linux" tab on cmake4eclipse settings page?

**Comment by 15knots** [comment]
One of the design goals of cmake4eclipse is to have Eclipse project files that work cross-platform.
Since Xcode is not available on Linux, it would be a bad idea to add that generator for Linux. We need an extra iOS (or OS X?) tab here.
Unfortunately I do not have anyy acces to iOS devicesto test it.
In case someone is plaaning to submit a pull request to solve this issue, I would happily provide guidance on how to do it.

**Comment by timursavchenko** [comment]
Hi! Yes, it's a better way to add a new tab called "macOS" and also add in dropdown list  generators for "Xcode", "Unix Makefiles" and "Ninja". So we'll have a possibility to build apps with help of Xcode generator for macOS and iOS targets. I could make all necessary changes in the code of cmake4eclipse, but I need help in plugin build and testing process, because I was unsuccessful with this last time I tried.

**Comment by 15knots** [comment]
Great if you want to help!
Just one question before you start: does the build-tool for xcode require some specific environment variables to be set. For example envvars that point to the include directories? If so, this might be a show-stopper since it will require an extra wrapper-script around the tool to set the envvars.

Concerning build and testing, see <https://github.com/15knots/cmake4eclipse#debug-and-build>

**Comment by timursavchenko** [comment]
No, it doesn't. The only thing I should have installed Xcode IDE or Xcode command line tools, but it's out of "cmake4eclipse" or Eclipse responsibility. The build process is similar to "Unix Makefiles" or "Ninja" generators: cmake -G"Xcode" and cmake --build . --config Release. After this commands cmake generates Xcode project files and run builder. Actually for macOS target I can use "Unix Makefiles" or "Ninja" generators as well. But not for iOS. Because of code signing stuff for it is only suitable Xcode generator.
I took a look at the "cmake4eclipse" source, and I think, it won't be an issue to implement Xcode generator support. The only problem for me is to build  "cmake4eclipse" and test it, because I have no experience with maven or Eclipse plugin development.
